### DISCUSSION TITLE
Massive Undertaking with MKDocs Material

### DISCUSSION THREAD

**Discussion by KeithIMyers** [squidfunk/mkdocs-material#7648] [Show and tell]
Hello All,
I am a huge fan of Google's ChromeOS and have released a series of books based on ChromeOS on several eBook retailers (The Ultimate ChromeOS Guide). Although these books are hardly best-sellers, I have received a lot of feedback from people who simply could not afford to buy the books as many were students or live in a location where the $10.00 pricetag is a small fortune. I can understand so I started working on making most of the content of my books available for free and when deciding on the platform I was going to use, I obviously went with MKDocs Material (Insiders). This project is still a work in progress and is already quite massive and I wanted to show a few things off. I am only about 40% completed with my final goal of making my content freely available to all.

The website is https://chromeOS.Guide and features the following

1) Hardware Documentation of over 650 ChromeOS Powered Devices (Still a WIP)
2) About 60% of the content of my books has been published on the "Web Guide"
3) Each ChromeOS Device has a custom built Chat Bot designed to help users get answers to their questions. Each chatbot was trained on a copy of the ChromeOS Guide specific to the device. It knows not to tell users to access features that are not included with the specific model. 
4) A Generic Chatbot is trained on the website and book.
5) There are over 2000 pages in the website that I have built

Just to give you a idea how large this website actually is, the output of the webroot is close to 2.2 GB, there are not a whole lot if images and all images are compressed webp images to minimize the amount of space needed.

Also, fun fact but I build and test everything related to the site on my Chromebook, MKDocs runs perfectly fine on here and my other custom build tools also run great.

**Comment by botkero** [comment]
Hi,

Awesome site!  
Do mind share some insight how you created the [timeline](https://chromeos.guide/History/ChromeOS-Device-History/)? Is that a plugin or self-made?

**Reply by kamilkrzyskow** [reply]
That's https://www.neoteroi.dev/mkdocs-plugins/timeline/ with custom adjustments.

**Reply by KeithIMyers** [reply]
> That's https://www.neoteroi.dev/mkdocs-plugins/timeline/ with custom adjustments.

Exactly, nothing too custom though. It's pretty vanilla TBH

**Reply by kamilkrzyskow** [reply]
Ah, I saw the custom images and a static red color, so I assumed you did something to keep the color static, but it turns out the timeline has so many elements that the gradient just shifts color a bit slower and I scrolled down too little. 

Over at Fiori Tracker the [history timeline](https://fioritracker.org/history/) is modified with custom CSS to keep the color static, I thought this case is similar 👌

**Reply by KeithIMyers** [reply]
> Ah, I saw the custom images and a static red color, so I assumed you did something to keep the color static, but it turns out the timeline has so many elements that the gradient just shifts color a bit slower and I scrolled down too little.
> 
> Over at Fiori Tracker the [history timeline](https://fioritracker.org/history/) is modified with custom CSS to keep the color static, I thought this case is similar 👌

Nope, that behavior is 100% vanilla. My customizations are mostly with the way the JSON file is generated prior to being fed into mkdocs
